src/main/java/seedu/duke/trackables/Deadline.java

```diff
@@ -0,0 +1,16 @@
+package seedu.duke.trackables;
+
+public class Deadline extends Task {
+
+    protected String by;
```

Comment:
Maybe you want to use "deadline" instead of "by" to improve the clarity of the code?